src/hotspot/share/gc/serial/markSweep.hpp

```diff
@@ -174,17 +170,13 @@ class MarkSweep : AllStatic {
   static void follow_array_chunk(objArrayOop array, int index);
 };
 
-class MarkAndPushClosure: public OopIterateClosure {
+class MarkAndPushClosure: public ClaimMetadataVisitingOopIterateClosure {
 public:
-  template <typename T> void do_oop_work(T* p);
-  virtual void do_oop(oop* p);
-  virtual void do_oop(narrowOop* p);
+  MarkAndPushClosure(int claim) : ClaimMetadataVisitingOopIterateClosure(claim) {}
```

Comment:
We always pass in the same value for `claim`. An alternative would be to remove this parameter and pass it to ClaimMetadataVisitingOopIterateClosure here. But maybe you liked having all the claim values together in one location in markSweep.cpp?